example/spring-data-mongodb/src/main/kotlin/com/github/inflab/example/spring/data/mongodb/repository/AddFieldsRepository.kt

```diff
@@ -0,0 +1,72 @@
+package com.github.inflab.example.spring.data.mongodb.repository
+
+import com.github.inflab.spring.data.mongodb.core.aggregation.aggregation
+import com.github.inflab.spring.data.mongodb.core.mapping.rangeTo
+import org.springframework.data.mongodb.core.MongoTemplate
+import org.springframework.data.mongodb.core.aggregate
+import org.springframework.data.mongodb.core.aggregation.AggregationResults
+import org.springframework.data.mongodb.core.mapping.Field
+import org.springframework.stereotype.Repository
+
+@Repository
+class AddFieldsRepository(
+    private val mongoTemplate: MongoTemplate,
+) {
+
+    data class Specs(val doors: Long?, val wheels: Long?, @Field("fuel_type") val fuelType: String?)
+    data class Vehicle(val id: Long, val type: String, val specs: Specs?)
+
+    data class Animal(val id: Long, val dogs: Long, val cats: Long)
+
+    data class Fruit(@Field("_id") val id: String, val item: String, val type: String)
```

Comment:
You can use `@Document` annotation, then it will be more convenient.

Reply:
Even when using the `@Document` annotation, it doesn't recognize 'id' as '_id'. I think it might be possible to achieve this by modifying the `toFieldName` according to the content of the following document.

https://docs.spring.io/spring-data/mongodb/docs/2.1.3.RELEASE/reference/html/#mongo-template.id-handling

Reply:
I think we can use `@id` annotation :) (same as remaining)

Reply:
Even when using the `@Id` anntation, it doesn't recognize 'id' as '_id'.

```kt
    fun overwriteAnExistingFieldWithField(): AggregationResults<Fruit> {
        val aggregation = aggregation {
            addFields {
                Fruit::id set Fruit::item
                Fruit::item set "fruit"
            }
        }

        return mongoTemplate.aggregate<Fruit>(aggregation, FRUITS)
    }
        }
```

Reply:
I understand, then you can update `toFieldName` to read `@id` annotation as document.

Reply:
Update `toFieldName` to check `@Id` annotation.

spring data monogo also has `_id` handling rule that a property or field without an annotation but named id maps to the _id field. [docs](https://docs.spring.io/spring-data/mongodb/docs/2.1.3.RELEASE/reference/html/#mongo-template.id-handling)

Shouldn't we add this rule?

Reply:
Yes, and there are some hardcoded "_id" literal in example repositories. 
you can update them too.